llvm/tools/sycl-post-link/ModuleSplitter.cpp

```diff
@@ -146,6 +146,9 @@ groupEntryPointsByKernelType(const ModuleDesc &MD,
     if (!isEntryPoint(F, EmitOnlyKernelsAsEntryPoints) ||
         !MD.isEntryPointCandidate(F))
       continue;
+    // Entry points should not be inlined
+    auto F1 = MD.getModule().getFunction(F.getName());
+    F1->addFnAttr(Attribute::NoInline);
```

Comment:
I don't think this is a good idea. Why are you adding this attribute here?

Reply:
Hi @bader 
Thanks for the comment.
As I mentioned earlier, the lower ESIMD pass adds 'always_inline' attribute to some functions. IT does not have information about entry points. Hence some entry points get marked with always_inline and gets removed by the AlwaysInliner pass that runs right after lower ESIMD pass. And we have a crash when we go looking for that entry point. 
Adding NoInline attribute prevents the lower ESIMD pass from marking 'entry point' functions with always_inline.
One other way I think that can work is to try and identify entry points in lower ESIMD pass and not mark them with always_inline.